src/subcommand/run.rs

```diff
@@ -21,47 +37,253 @@ pub enum Error {
+            warnings in the logs for more details.
+            "})?;
+
+        // Filter to the list of references that are new since we last saw them.
+        let references = futures::stream::iter(references.into_iter())
+            // Using `filter_map` instead of `filter` so that this closure gets ownership of `reference`,
+            // which makes binding it across an await point easier (no lifetimes to mess with).
+            .filter_map(|reference| async {
+                let coordinate = reference.as_coordinate(&remote);
+                match db.state(&coordinate).await {
+                    // No previous state; this must be a new reference.
+                    Ok(None) => Some(Ok(reference)),
+                    // There was previous state, it's only new if the state is different.
+                    // We're assuming "different state" always means "newer state". 
+                    // This is because state is currently expressed as a git commit string,
+                    // which on its own doesn't have any form of ordering.
+                    Ok(Some(db_state)) => {
+                        if db_state == reference.as_state() {
+                            None
+                        } else {
+                            Some(Ok(reference))
+                        }
+                    }
+                    // Pass through errors.
+                    Err(err) => Some(Err(err)),
+                }
+            })
+            // Fold ourselves because `collect` expects its destination to implement `Default`.
+            .try_fold(Vec::new(), |mut references, reference| async {
+                references.push(reference);
+                Ok(references)
+            })
+            .await
+            .change_context(Error::PollIntegration)
+            .describe_lazy(|| {
+                format!("filter to only changes at {remote} in integration: {integration}")
+            })
+            .help(indoc! {"
+            Problems at this stage are most likely caused by a database error.
+            Broker manages a local sqlite database; deleting it so it can be re-generated from scratch may resolve the issue.
+            "})?;
+
+        // We sink the references here instead of during the stream so that
+        // if an error is encountered reading the stream, we don't send partial lists.
+        if references.is_empty() {
+            info!("No changes to {integration} since last poll interval");
+        }
+        for reference in references {
+            let job = ScanGitVCSReference::new(integration, &reference);
+            let encoded = bincode::serialize(&job).context(Error::TaskEncode)?;
+            let mut sender = sender.lock().await;
+            sender
+                .send(&encoded)
+                .await
+                .change_context(Error::TaskEnqueue)?;
+
+            info!("Enqueued task to scan {integration} at {reference}");
```

Comment:
I found these a bit hard to parse. Here's what they currently look like:

```
Enqueued task to scan using git protocol with transport cloning via SSH from git@github.com:fossas/broker.git with key file at "/Users/scott/.ssh/id_ed25519" at git branch 'test-with-async_with_vars' with head commit '98f35745378eaaf375b0df6e055e7166077c677f'
```

I tried two things to see if I could improve this, neither of which is perfect.

First, I just wrapped some phrases in quotes to group them:

```
Enqueued task "scan using git protocol" with transport "cloning via SSH" from git@github.com:fossas/broker.git with key file at "/Users/scott/.ssh/id_ed25519" at git branch 'test-with-async_with_vars' with head commit '98f35745378eaaf375b0df6e055e7166077c677f'
```

I also tried making things more "JSONy" like this, but this might be deep in the uncanny valley:

```
Enqueuing task: "scan using git protocol", transport: "cloning via SSH", remote: git@github.com:fossas/broker.git, auth: "key file at '/Users/scott/.ssh/id_ed25519'", reference: "git branch 'test-with-async_with_vars' with head commit '98f35745378eaaf375b0df6e055e7166077c677f'"
```

Reply:
yeah, this is the "the wording in these info messages is really verbose, I'll probably take a pass at displaying these in a more useful way to the user".

I'd like to leave it here for now but with the knowledge/hope that we'll go touch these up. I am not yet certain how to make them useful for the user but without making them literal json-style outputs (which semi-technical users will find hard to parse as well). I suspect that if we address the "the code to keep this functionality generalized is starting to spaghettify pretty badly, if I get time I may take a redesign pass on how we organize our types & modules" issue this may get easier as well.